src/lib/logical_query_plan/jit_aware_lqp_translator.cpp

```diff
@@ -0,0 +1,294 @@
+         filter_node->type() != LQPNodeType::Union) {
+    filter_node = filter_node->left_input();
+  }
+
+  // If we can reach the input node without encountering a UnionNode or PredicateNode,
+  // there is no need to filter any tuples
+  if (filter_node != input_node) {
+    // However, if we do need to filter, we first convert the filter node to a JitExpression ...
+    const auto expression = _translate_node_to_jit_expression(filter_node, *read_tuple, input_node);
+    if (!expression) return nullptr;
+    // make sure that the expression gets computed ...
+    jit_operator->add_jit_operator(std::make_shared<JitCompute>(expression));
+    // and then filter on the resulting boolean.
+    jit_operator->add_jit_operator(std::make_shared<JitFilter>(expression->result()));
+  }
+
+  // Identify the top-most projection node to determine the output columns
+  auto top_projection = node;
+  while (top_projection != input_node && top_projection->type() != LQPNodeType::Projection) {
+    top_projection = top_projection->left_input();
+  }
+
+  // Add a compute operator for each computed output column (i.e., a column that is not from a stored table).
+  auto write_table = std::make_shared<JitWriteTuple>();
+  for (const auto& output_column :
+       boost::combine(top_projection->output_column_names(), top_projection->output_column_references())) {
+    const auto expression = _translate_column_to_jit_expression(output_column.get<1>(), *read_tuple, input_node);
+    if (!expression) return nullptr;
+    // It the JitExpression is of type ExpressionType::Column, there is no need to add a compute node, since it
+    // would not compute anything anyway
+    if (expression->expression_type() != ExpressionType::Column) {
+      jit_operator->add_jit_operator(std::make_shared<JitCompute>(expression));
+    }
+    write_table->add_output_column(output_column.get<0>(), expression->result());
+  }
+  jit_operator->add_jit_operator(write_table);
+  return jit_operator;
+}
+
+std::shared_ptr<const JitExpression> JitAwareLQPTranslator::_translate_node_to_jit_expression(
+    const std::shared_ptr<AbstractLQPNode>& node, JitReadTuple& jit_source,
+    const std::shared_ptr<AbstractLQPNode>& input_node) const {
+  std::shared_ptr<const JitExpression> left, right;
+  switch (node->type()) {
+    case LQPNodeType::Predicate:
+      // If the PredicateNode has further conditions (either PredicateNodes or UnionNodes) as children, we need a
+      // logical AND here. In this case, the condition represented by the PredicateNode itself becomes the left side
+      // and the input to the predicate node becomes the right side of the conjunction.
+      // If the PredicateNode has no further conditions (i.e., all tuples pass into the PredicateNode unconditionally),
+      // there is no need to create a dedicated AND-expression.
+      if (_subquery_filters_tuples(node->left_input())) {
```

Comment:
Did we have that in the version that I reviewed as well? I think it is confusing as we already have "subqueries" in SQL, and they are something different. Maybe `_input_is_filtered`?